clang-tidy errors (use after move)

After adding the ConcurrentQueue to HPX we see some clang-tidy errors 

They appear to be benign, caused by the slightly tricky code that calls the destructor if the move fails.
I have added `// NOLINT` to lines locally to suppress the errors, but I open this issue just to alert you to their existence in case you wish to look into it. Please close this issue if you are already aware of the problem.

Please note that the line numbers might be offset relative to your version. I removed our extra header lines to generate the errors here but I may be 1 or 2 lines out.
```
clang-tidy src/runtime/threads/executors/this_thread_executors.cpp -- -I/home/biddisco/src/hpx-master -I/home/biddisco/build/hpx-master -I ~/apps/boost/1.68.0/include/ -I ~/apps/hwloc/2.1.0/include/ -DDEBUG -DHPX_COMPONENT_NAME=hpx -DHPX_ENABLE_ASSERT_HANDLER -DHPX_EXPORTS -D_DEBUG -D_GNU_SOURCE -Dhpx_EXPORTS  -fcolor-diagnostics -g -fPIC   -pthread -Wall -Wextra -Wno-unused-local-typedefs -Wno-strict-aliasing -Wno-sign-promo -Wno-attributes -Wno-cast-align -Wno-unused-parameter -Wformat=2 -Wno-format-nonliteral -Winit-self -Wdouble-promotion -Wcast-qual -Werror=parentheses -Werror=reorder -Werror=return-type -Werror=sequence-point -Werror=uninitialized -Werror=format -Werror=missing-braces -Werror=sign-compare -fdiagnostics-show-option -Werror=vla -std=c++17
1085 warnings generated.
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:1966:45: error: 'el' used after it was moved [bugprone-use-after-move,-warnings-as-errors]
                        element = std::move(el);
                                            ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:1952:69: note: move occurred here
                    if (!MOODYCAMEL_NOEXCEPT_ASSIGN(T, T&&, element = std::move(el))) {
                                                                    ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:1966:45: note: the use happens in a later loop iteration than the move
                        element = std::move(el);
                                            ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:1969:45: error: 'el' used after it was moved [bugprone-use-after-move,-warnings-as-errors]
                        element = std::move(el);
                                            ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:1952:69: note: move occurred here
                    if (!MOODYCAMEL_NOEXCEPT_ASSIGN(T, T&&, element = std::move(el))) {
                                                                    ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:1969:45: note: the use happens in a later loop iteration than the move
                        element = std::move(el);
                                            ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:1970:25: error: 'el' used after it was moved [bugprone-use-after-move,-warnings-as-errors]
                        el.~T();
                        ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:1969:33: note: move occurred here
                        element = std::move(el);
                                ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:2516:45: error: 'el' used after it was moved [bugprone-use-after-move,-warnings-as-errors]
                        element = std::move(el);
                                            ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:2494:69: note: move occurred here
                    if (!MOODYCAMEL_NOEXCEPT_ASSIGN(T, T&&, element = std::move(el))) {
                                                                    ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:2516:45: note: the use happens in a later loop iteration than the move
                        element = std::move(el);
                                            ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:2519:45: error: 'el' used after it was moved [bugprone-use-after-move,-warnings-as-errors]
                        element = std::move(el);
                                            ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:2494:69: note: move occurred here
                    if (!MOODYCAMEL_NOEXCEPT_ASSIGN(T, T&&, element = std::move(el))) {
                                                                    ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:2519:45: note: the use happens in a later loop iteration than the move
                        element = std::move(el);
                                            ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:2520:25: error: 'el' used after it was moved [bugprone-use-after-move,-warnings-as-errors]
                        el.~T();
                        ^
/home/biddisco/src/hpx-master/hpx/util/lockfree/concurrentqueue.hpp:2519:33: note: move occurred here
                        element = std::move(el);
                                ^
Suppressed 1077 warnings (1075 in non-user code, 2 with check filters).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
6 warnings treated as errors
```

--- a/concurrentqueue.h
+++ b/concurrentqueue.h
@@ -1962,12 +1962,12 @@ private:
 								block->ConcurrentQueue::Block::template set_empty<explicit_context>(index);
 							}
 						} guard = { block, index };
-						
-						element = std::move(el);
+
+						element = std::move(el); // NOLINT
 					}
 					else {
-						element = std::move(el);
-						el.~T();
+						element = std::move(el); // NOLINT
+						el.~T(); // NOLINT
 						block->ConcurrentQueue::Block::template set_empty<explicit_context>(index);
 					}
 					
@@ -2512,13 +2512,13 @@ private:
 								}
 							}
 						} guard = { block, index, entry, this->parent };
-						
-						element = std::move(el);
+
+						element = std::move(el); // NOLINT
 					}
 					else {
-						element = std::move(el);
-						el.~T();
-					
+						element = std::move(el); // NOLINT
+						el.~T(); // NOLINT
+
 						if (block->ConcurrentQueue::Block::template set_empty<implicit_context>(index)) {
 							{
 #if MCDBGQ_NOLOCKFREE_IMPLICITPRODBLOCKINDEX

